### ISSUE TITLE
Deadlock on User Navigation if I/O in the onNav method

### ISSUE THREAD

**Issue by jwmach1** [maxence-charriere/go-app#368]
I realized I was getting lucky on when I performed I/O.  I have been calling back to my server for some REST endpoints.  Everything was ok, calling the server as a result of 
* initially navigating to my "main" component mounted at "/" (LIST endpoint)
* calling the server with a CREATE after the user clicks a button on a "/create" component

But upon the user navigating back (browser back button), the WASM would crash with the deadlock as the UI thread was blocked.  Taking a look at `app_wasm.go` I see (code below from SHA be5607) that the onPopState method dispatches the navigate call on the UI channel.

```go
func onPopState(this Value, args []Value) interface{} {
	if u := Window().URL(); u.Fragment == "" {
		dispatcher(func() {
			navigate(u, false)
		})
	}
	return nil
}
```

This might be a bug against the onPopState, or I might need to make sure, for myself as implementer that I/O should be performed in my own goroutine. if the later is true, perhaps some documentation for guidance where I/O will lead to deadlock. I can likely contribute _something_ here.

**Comment by maxence-charriere** [comment]
I made the on pop state be called on the UI goroutine because I want the global behavior that each callback created by the app package to be executed on the UI goroutine.

I/O are blocking calls so if you are doing directly, it likely gonna block the UI.
This is not a bug.

In the previous versions, I did a system similar to flux to handle this but it was not obvious for everyone so I decided to remove it and let users choose what concurrency model they want.
